### PR TITLE
Made a giveaway command

### PR DESCRIPTION
**Pull request by RopraMMC**
(: even caeden likes it so it is good (although he bullied me)

### CODE REVIEW

**Comment by Dekriel** [comment]
what happens in discord stays in discord.

**Comment by RopraMMC** [comment]
note: intents but be all

**Comment by Yat-o** [comment]
Dude wtf did you seriously add asyncio to the requirements file

File: src/cogs/misc.py [hunk 1]
```diff
@@ -149,6 +152,52 @@ async def report(self, ctx: Context):
         embed = EmbedFactory.ide_embed(ctx, bug_string.format(bug_id))
         await ctx.send(embed=embed)
 
+@command(aliases=["give", "gaway", "gcreate"])
+@commands.has_permissions(manage_messages=True)
+async def giveaway(self, ctx: Context, winners: int, time: str, *, prize: str):
+    """Allows you to giveaway prizes."""
+
+    winnerList = []
```
Comment (Yat-o) [inline src/cogs/misc.py:160]: camel case is not pep8 compliant

File: src/cogs/misc.py [hunk 2]
```diff
@@ -149,6 +152,52 @@ async def report(self, ctx: Context):
...
+        text = "Press the :tada: emoji to enter!",
+    ))
+
+    await giveawayMessage.add_reaction("🎉")
+
+    time_convert = {"s":1, "m":60, "h":3600, "d":8600}
+
+    giveawayTime = int(time[:-1]) * time_convert[time[-1]]
+
+    await asyncio.sleep(giveawayTime)
```
Comment (Yat-o) [inline src/cogs/misc.py:177]: sleeping is a bad way to handle giveaways. What if the bot goes offline. The state of the sleep isn’t saved.

**Review by Yat-o — CHANGES_REQUESTED** [review]

**Comment by Yat-o** [comment]
There’s some more stuff but you know. You’ll get those.

File: src/cogs/misc.py (hunk 2 above)
Comment (realstealthninja) [reply src/cogs/misc.py:177]: i have to agree with yat-o here alot of ways this could fail

File: src/cogs/misc.py [hunk 3]
```diff
@@ -149,6 +152,52 @@ async def report(self, ctx: Context):
...
+
+    newGiveawayMessage = await ctx.channel.fetch_message(giveawayMessage.id)
+
+
+    users = await newGiveawayMessage.reactions[0].users().flatten()
+    users.pop(users.index(self.bot.user))
```
Comment (Yat-o) [inline src/cogs/misc.py:183]: You don’t even need to do this. Just use list comprehension when you make the initial list. 

```py
[user for user in message.reactions[0].users().flatten() if not user.bot]
```
Something along those lines

**Review by realstealthninja — CHANGES_REQUESTED** [review]
needs a way to save time

File: src/cogs/misc.py (hunk 1 above)
Comment (RopraMMC) [reply src/cogs/misc.py:160]: HSIT I FORGT TO CHANGE THEM
SHIT

**Comment by RopraSchool** [comment]
Yo imm gonna close this when I get home, I am going to start edititing from school aswell

File: src/cogs/misc.py (hunk 2 above)
Comment (RopraSchool) [reply src/cogs/misc.py]: I can't be asked to fix it so im going to just going to close when im home

**Pull request closed without merging**